### PR TITLE
[#27] [Android] [UI] As a user, I can see many question types: nps rating

### LINKED ISSUE #27: [Android] [UI] As a user, I can see many question types: nps rating
## Why

The survey allows the user to fill in the answer in different ways: rating bar, choice, etc
 
## Acceptance Criteria

- number rating

### PR DESCRIPTION
**Pull request by luongvo**
- Close #27

## What happened 👀

Implement the `Nps` selector composable to produce the display type: nps.

## Insight 📝

- Sync all the latest changes in AppTheme management from [Android Templates](https://github.com/nimblehq/android-templates/pull/376/files#diff-25c2cf05d5f277d83a6f9cd73526a6acf41506b09906e00cd06419f5b38469b8).
- Make use of [Compose Constraint Layout](https://developer.android.com/jetpack/compose/layouts/constraintlayout) to have better layout constraints.

## Proof Of Work 📹

https://user-images.githubusercontent.com/16315358/217613967-d827db5c-68eb-4e51-91d0-b7b9ab535cce.mp4

```
2023-02-09 00:48:44.103 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 1
2023-02-09 00:48:44.702 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 3
2023-02-09 00:48:45.065 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 5
2023-02-09 00:48:45.732 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 7
2023-02-09 00:48:46.112 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 8
2023-02-09 00:48:46.449 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 9
2023-02-09 00:48:46.955 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 2
2023-02-09 00:48:47.462 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 6
2023-02-09 00:48:47.732 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 7
2023-02-09 00:48:48.001 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 9
2023-02-09 00:48:48.289 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 9
2023-02-09 00:48:48.720 15463-15463 SurveyQues...AnswerForm vn....vo.kmm.survey.android.staging  D  NPS -> onValueChange: 10
```

### CODE REVIEW

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/survey/views/Nps.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import androidx.constraintlayout.compose.Dimension
+import vn.luongvo.kmm.survey.android.R
+import vn.luongvo.kmm.survey.android.ui.screens.survey.AnswerUiModel
+import vn.luongvo.kmm.survey.android.ui.theme.AppTheme.dimensions
+import vn.luongvo.kmm.survey.android.ui.theme.AppTheme.shapes
+import vn.luongvo.kmm.survey.android.ui.theme.AppTheme.typography
+import vn.luongvo.kmm.survey.android.ui.theme.White50
+
+@Composable
+fun Nps(
```
[the remark this answers is not in the document]
Comment (blyscuit) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/survey/views/Nps.kt:25]: Please suppress.

File: android/build.gradle.kts [hunk 2]
```diff
@@ -119,11 +119,12 @@ dependencies {
...
         implementation(NAVIGATION)
         implementation(UI_TOOLING)
-        implementation(COIL_COMPOSE)
+
         implementation(ACCOMPANIST_PAGER)
         implementation(ACCOMPANIST_PAGER_INDICATORS)
         implementation(ACCOMPANIST_PLACEHOLDER)
-
+        implementation(COIL_COMPOSE)
+        implementation(CONSTRAINT_LAYOUT)
```
Comment (blyscuit) [inline android/build.gradle.kts:127]: Curious, this plugin is for getting the screen size?

Comment (luongvo) [reply android/build.gradle.kts:127]: @blyscuit [That's a more powerful Layout in Android](https://developer.android.com/jetpack/compose/layouts/constraintlayout) for both XML and Jetpack Compose. It combines the features of Row, Column, Box, etc with constraint attrs. 🤓

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/survey/views/Nps.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/survey/views/Nps.kt:25]: Fixed in https://github.com/luongvo/kmm-survey/pull/79/commits/a8aa44763d935561c2e187450604889b07fd843c

**Review by blyscuit — APPROVED** [review]
What happened to your codcov? 🙈

**Comment by luongvo** [comment]
> What happened to your codcov? 🙈

@blyscuit More UI code has been added without more tests being covered yet. I will take care of adding tests at the integration step.

**Review by suho — APPROVED** [review]

**Review by doannimble — APPROVED** [review]
LGTM 💯

**Pull request merged**